cmd/sonobuoy/app/images.go

```diff
@@ -0,0 +1,293 @@
+/*
+Copyright 2019 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"syscall"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/docker/docker/pkg/term"
+	"github.com/heptio/sonobuoy/pkg/errlog"
+	"github.com/heptio/sonobuoy/pkg/image"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"golang.org/x/crypto/ssh/terminal"
+	"golang.org/x/net/context"
+)
+
+var imagesflags imagesFlags
+
+type imagesFlags struct {
+	e2eRegistryConfig string
+	plugin            string
+	username          string
+	password          string
+	kubeconfig        Kubeconfig
+
+	imagesflags *pflag.FlagSet
+}
+
+func ImagesFlagSet(cfg *imagesFlags) *pflag.FlagSet {
+	flagset := pflag.NewFlagSet("images", pflag.ExitOnError)
+	AddPluginFlag(&cfg.plugin, flagset)
+	AddE2ERegistryConfigFlag(&cfg.e2eRegistryConfig, flagset)
+	AddRegistryUsernameFlag(&cfg.username, flagset)
+	AddKubeconfigFlag(&cfg.kubeconfig, flagset)
+	cfg.imagesflags = flagset
+	return flagset
+}
+
+func NewCmdImages() *cobra.Command {
+	// Main command
+	cmd := &cobra.Command{
+		Use:   "images",
+		Short: "Manage images used in a plugin. Supported plugins are: 'e2e'",
+		Run:   getImages,
+		Args:  cobra.ExactArgs(0),
+	}
+
+	cmd.Flags().AddFlagSet(ImagesFlagSet(&imagesflags))
+	cmd.MarkFlagRequired(pluginFlag)
+
+	// Sub command
+	subCmd := &cobra.Command{
+		Use:   "download",
+		Short: "Download images to local Docker client for a specific plugin",
+		Run:   downloadImages,
+		Args:  cobra.ExactArgs(0),
+	}
+	subCmd.Flags().AddFlagSet(ImagesFlagSet(&imagesflags))
+	subCmd.MarkFlagRequired(pluginFlag)
+	subCmd.MarkFlagRequired(e2eRegistryConfigFlag)
```

Comment:
For downloading they shouldn't need this; for the upload it makes sense though.

Reply:
Depends on if we change the functionality. I don't think most air-gapped environments are truly air-gapped, they just don't have internet access.

Reply:
But if you can' reach the internet you _cant_ do the download/upload at the same time and still have to walk it to the other network? In some cases you do have one particular machine that can reach both, but I guess I dont have the experience to know which topology is more common.